### PR TITLE
Implementation of the Profile-Payment Integration

### PR DESCRIPTION
**Pull request by BerryGIT-ME**
### What this PR does (required):
- Completes the payment feature and makes it possible to add payment cards and make payments from the UI

#### Adding a payment method

https://user-images.githubusercontent.com/53505772/136446113-b602aae0-5c41-4343-b66a-847b4bdf8371.mp4

#### Confirming or canceling a payment

https://user-images.githubusercontent.com/53505772/136446227-b3422300-4aaa-4cc3-a077-bdf69c750cbd.mp4

### Any information needed to test this feature (required):
- In the server folder run `node seed/index.js`  to populate the database with mock data
- log in using the credentials of any dog owner in the mock data
- navigate to this endpoint `/profile-settings/payments` 

### Any issues with the current functionality (optional):
- A bug in the useAuthContext.tsx file causes the app to malfunction. To test the feature simply comment out the useEffects in the file after log-in
- the bugFix worked on by @mayanksuchde [can be found here](https://github.com/hatchways/team-brioche/pull/80#issue-1019560977)

### CODE REVIEW

File: server/seedPayment/index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+require("dotenv").config();
+const colors = require("colors");
+const User = require("../models/User");
+const Profile = require("../models/Profile");
+const { userArray, profileArray } = require("./sampleProfles");
+const connectDB = require("../db");
+
+connectDB();
+
+console.log("loading mock users");
```
Comment (ezeilo-su) [inline server/seedPayment/index.js:10]: Remove debug console log, please.

File: server/seedPayment/index.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    const userId = users.map((user) => user._id);
+    let index = 0;
+
+    for (let profile of profileArray) {
+      profile.userId = userId[index];
+      await Profile.create(profile);
+      index++;
+    }
+    console.log("User ID's: ", userId);
+    console.log("populating profiles...");
```
Comment (ezeilo-su) [inline server/seedPayment/index.js:28]: Remove debug console logs, please.

File: server/seedPayment/index.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        email: 1,
+      });
+      console.log(profiles);
+      console.log(`
+      seeding complete 
+      summary: 
+        Number of users created : ${users.length}
+        Number of profiles created : ${profiles.length}
+        Exiting process
+      `);
```
Comment (ezeilo-su) [inline server/seedPayment/index.js:42]: Remove debug console logs, please.

**Review by rajivtitus — APPROVED** [review]
LGTM 👍🏽

**Comment by BerryGIT-ME** [comment]
@rajivtitus this is because I commented out the AuthProvider after logging in as a user. I gave the reason for that in the description of this PR.

**Pull request merged**